submissions/models.py

```diff
@@ -33,6 +33,11 @@
     providing_args=['anonymous_user_id', 'course_id', 'item_id']
 )
 
+# In order to allow for effective soft-deletes, we want to override the default Manager for several object types
+class SoftDeletedManager(models.Manager):
+    def get_queryset(self):
+        #By default, filter out any soft-deleted records for classes using this manager
+        return super(SoftDeletedManager, self).get_queryset().exclude(deleted=True)
```

Comment:
@maxrothman, could we chat sometime about the performance implication of this approach? We're effectively adding a `WHERE NOT ('submissions_submission'.'deleted' = True)` clause to every query django makes to the `submissions` table. We're expecting a very large percentage of items to have `deleted` be False, the only ones that will be True are those the the instructor has individually chosen for use with the `Delete Student State` tool.

Does this set your devops-spidey senses tingling? Is there a way we could do this more intelligently?

Ticket for reference: https://openedx.atlassian.net/browse/TNL-3880